Core/PARStore.m

```diff
@@ -949,6 +950,85 @@ - (void)setEntriesFromDictionary:(NSDictionary *)dictionary
      }];
 }
 
+- (void)appendChanges:(NSArray *)changes forDeviceIdentifier:(NSString *)deviceIdentifier
+{
+    // Model and PSC
+    NSManagedObjectModel *mom = [PARStore managedObjectModel];
+    NSPersistentStoreCoordinator *psc = [[NSPersistentStoreCoordinator alloc] initWithManagedObjectModel:mom];
+    
+    // Persistent Store
+    NSError *error = nil;
+    NSString *dirPath = [self directoryPathForDeviceIdentifier:deviceIdentifier];
+    [[NSFileManager defaultManager] createDirectoryAtPath:dirPath withIntermediateDirectories:NO attributes:nil error:NULL];
+    [self addPersistentStoreWithCoordinator:psc dirPath:dirPath readOnly:NO error:&error];
+    
+    // Context
+    NSManagedObjectContext *moc = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSPrivateQueueConcurrencyType];
+    [moc setPersistentStoreCoordinator:psc];
+    [moc setUndoManager:nil];
+
+    // Have main context observe changes made in background, and merge them.
+    id observer = [[NSNotificationCenter defaultCenter] addObserverForName:NSManagedObjectContextDidSaveNotification object:moc queue:nil usingBlock:^(NSNotification *notification) {
+        [self.databaseQueue dispatchSynchronously:^{
+            [[self managedObjectContext] mergeChangesFromContextDidSaveNotification:notification];
+        }];
+    }];
+    
+    // Add changes
+    [moc performBlockAndWait:^{
+        NSError *error = nil;
+        
+        // Expression to retrieve the maximum timestamp in the store
```

Comment:
My main concern had to do with the fact we have overlapping change sets and we don't want to reinsert the same row twice, the timestamp implementation (without checking the property) is not suitable for this (the querying without checking the data blob isn't 100% either, but much safe, i.e. need to have at least 3 changes of the same property with the same timestamp before getting into trouble).

Re: timestamp, if we indeed cannot rely on the last row inserted is what we need, than @max is the right choice (however moot as we need to move to querying).